### PR TITLE
Add resources on APIs

### PR DESCRIPTION
**Pull request by JennyBrennan**
I recently found this list of free APIs which I find really useful in teaching coding, and thought would be useful for self-teachers too! Also added an article from CodeNewbie on introducing APIs to give context for learners. Awesome repo Charlotte 🙌

### CODE REVIEW

File: online-resources/learn-to-code-resources.md [hunk 1]
```diff
@@ -164,8 +164,12 @@
 
 * [You Learn Haskell for Great Good](http://learnyouahaskell.com/chapters)
 
-
-##Free Lynda Account
+## Free Lynda Account
```
Comment (JennyBrennan) [inline online-resources/learn-to-code-resources.md]: This is a drive-by, but whilst I was here I thought I'd fix the markdown on the entry above 😄

Comment (charlottebrf) [reply online-resources/learn-to-code-resources.md]: Thank you! Most appreciated 💃

File: online-resources/learn-to-code-resources.md [hunk 2]
```diff
@@ -164,8 +164,12 @@
...
 Most public libraries have a connection with an affiliate lynda This relationship allows for public library card holder to get free account with lynda. Lynda is an online resource that has course that cover most things in the coding world
 
 * [Lynda](https://www.lynda.com)
+
+## APIs
+
```
Comment (charlottebrf) [inline online-resources/learn-to-code-resources.md]: Excellent addition!
